### DISCUSSION TITLE
Using AppRole, SecretId and Namespace with hvac client

### DISCUSSION THREAD

**Discussion by wandermonk** [hvac/hvac#754] [Q&A]
Hi Team,

I am trying to use HVAC to fetch the token and read the creds stored in Vault. But, I am finding it very difficult to get hands-on the appropriate documentation. Therefore, I am opening this issue. Any help is highly appreciated.

How to achieve the below functionality using HVAC library? I am using the requests library but now I would like to use the HVAC library instead.

Please find the code here

https://gist.github.com/wandermonk/c2efcea841f807e151fd4c1be4ef3b04

**Comment by CTinMich** [comment]
Hello WanderMonk...  I am struggling too...  I ended up with a half-and-half solution for us.  The routine to get the "secret_id" was done with "requests" (I still can not figure out the HVAC commands to do this even though it "looks like" it should be simple...  The routine to read from the vault I do have using the HVAC commands.  Please let me know what you figure out because I want an "all" HVAC solution also.
[VaultCall_example.txt](https://github.com/hvac/hvac/files/7032881/VaultCall_example.txt)
